### PR TITLE
[86bye3tve][accordion] added use prop

### PR DESCRIPTION
**Pull request by msereniti**
## Motivation and Context

For a long time our design has both compact and non compact accordions. Alongside react component were not supporting non compact accordion without hacks. This PR adds simple `compact` prop that is `true` by default for backward capability.

## How has this been tested?

Manually and with VR.

## Screenshots (if appropriate):

Screenshot 2024-06-27 at 17 43 14

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly or it's not required.
- [x] Unit tests are not broken.
- [x] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [x] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

File: semcore/accordion/src/index.d.ts [hunk 1]
```diff
@@ -52,6 +58,10 @@ export type AccordionItemProps = {
   /** Animation duration
    * @default 350 */
   duration?: number;
+  /**
+   * Changes the visual appearance of the item
+   */
+  compact?: boolean;
```
Comment (ilyabrower) [inline semcore/accordion/src/index.d.ts:64]: not sure, that this is public property... I think all items in Accordion should be the same

File: semcore/accordion/src/index.d.ts [hunk 2]
```diff
@@ -70,6 +80,10 @@ export type ChevronItemProps = BoxProps & {
    * @default m
    */
   size?: 'm' | 'l';
+  /**
+   * Changes the visual appearance of the item
+   */
+  compact?: boolean;
```
Comment (ilyabrower) [inline semcore/accordion/src/index.d.ts:86]: and here too

File: semcore/accordion/src/style/accordion.shadow.css [hunk 3]
```diff
@@ -14,6 +14,14 @@ SItemToggle[keyboardFocused]:not([tabIndex='-1'])::after {
   z-index: 3;
 }
 
+SItemToggle:not(SItemToggle[compact]) {
+  background-color: var(--intergalactic-bg-secondary-neutral, #f4f5f9);
+  padding: var(--intergalactic-spacing-2x, 8px) var(--intergalactic-spacing-3x, 12px);
+}
+SItemToggle:not(SItemToggle[compact]):hover {
+  background-color: var(--intergalactic-bg-secondary-neutral-hover, #e0e1e9);
+}
```
Comment (ilyabrower) [inline semcore/accordion/src/style/accordion.shadow.css:23]: What do you think about adding the ability to set a custom background color?

Comment (msereniti) [reply semcore/accordion/src/style/accordion.shadow.css:23]: We should discuss this feature

**Review by ilyabrower — APPROVED** [review]

**Comment by j-mnizhek** [comment]
1. This accordion should be `secondary` not `primary`, since it usually uses less important place in the interface hierarchy.
image

2. Can we add `margin-bottom: var(--intergalactic-spacing-05x)` (or gap) by default for the ItemToggle for the primary accordion (as it was in the design)? The last child of course shouldn't have it.
image

**Review by j-mnizhek — CHANGES_REQUESTED** [review]
Suggested some changes in the comment to the PR.

**Comment by ilyabrower** [comment]
> 2. Can we add `margin-bottom: var(--intergalactic-spacing-05x)` (or gap) by default for the ItemToggle for the primary accordion (as it was in the design)? The last child of course shouldn't have it.
I've fixed it in https://github.com/semrush/intergalactic/pull/1468

**Review by j-mnizhek — APPROVED** [review]

**Pull request merged**